File: invesdwin-scripting-parent/invesdwin-scripting-haskell/invesdwin-scripting-haskell-runtime-contract/src/test/java/de/invesdwin/scripting/haskell/runtime/contract/callback/ParametersAndReturnsTestInteger.hs
println("getInteger")
if isdefined(Main, :getInteger) && !isnothing(getInteger)
	error("getInteger already defined!")
end
getInteger = callback("getInteger")
println(typeof(getInteger))
println(getInteger)
if typeof(getInteger) != Int32
	error("getInteger not Int32!")
end
callback("setInteger",getInteger)

println("getIntegerVector")
if isdefined(Main, :getIntegerVector) && !isnothing(getIntegerVector)
	error("getIntegerVector already defined!")
end
getIntegerVector = callback("getIntegerVector")
println(eltype(getIntegerVector))
println(getIntegerVector)
if eltype(getIntegerVector) != Int32
	error("getIntegerVector not Int32!")
end
callback("setIntegerVector",getIntegerVector)

println("getIntegerVectorAsList")
if isdefined(Main, :getIntegerVectorAsList) && !isnothing(getIntegerVectorAsList)
	error("getIntegerVectorAsList already defined!")
end
getIntegerVectorAsList = callback("getIntegerVectorAsList")
println(eltype(getIntegerVectorAsList))
println(getIntegerVectorAsList)
if eltype(getIntegerVectorAsList) != Int32
	error("getIntegerVectorAsList not Int32!")
end
callback("setIntegerVectorAsList",getIntegerVectorAsList)

println("getIntegerMatrix")
if isdefined(Main, :getIntegerMatrix) && !isnothing(getIntegerMatrix)
	error("getIntegerMatrix already defined!")
end
getIntegerMatrix = callback("getIntegerMatrix")
println(eltype(getIntegerMatrix))
println(getIntegerMatrix)
if eltype(getIntegerMatrix) != Int32
	error("getIntegerMatrix not Int32!")
end
callback("setIntegerMatrix",getIntegerMatrix)

println("getIntegerMatrixAsList")
if isdefined(Main, :getIntegerMatrixAsList) && !isnothing(getIntegerMatrixAsList)
	error("getIntegerMatrixAsList already defined!")
end
getIntegerMatrixAsList = callback("getIntegerMatrixAsList")
println(eltype(getIntegerMatrixAsList))
println(getIntegerMatrixAsList)
if eltype(getIntegerMatrixAsList) != Int32
	error("getIntegerMatrixAsList not Int32!")
end
callback("setIntegerMatrixAsList",getIntegerMatrixAsList)

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/callback/ParametersAndReturnsTestInteger.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getInteger")
if(getInteger ~= nil) then
	error("getInteger already defined!")
end
getInteger = callback("getInteger")
getIntegerType = type(getInteger)
print(getIntegerType)
print(getInteger)
if(getIntegerType ~= "number") then
	error("getInteger not Integer!")
end
callback("setInteger", getInteger)

print("getIntegerVector")
if(getIntegerVector ~= nil) then
	error("getIntegerVector already defined!")
end
getIntegerVector = callback("getIntegerVector")
getIntegerVectorType = type(getIntegerVector[1])
print(getIntegerVectorType)
print(arraysToString(getIntegerVector))
if(getIntegerVectorType ~= "number") then
	error("getIntegerVector not Integer!")
end
callback("setIntegerVector", getIntegerVector)

print("getIntegerVectorAsList")
if(getIntegerVectorAsList ~= nil) then
	error("getIntegerVectorAsList already defined!")
end
getIntegerVectorAsList = callback("getIntegerVectorAsList")
getIntegerVectorAsListType = type(getIntegerVectorAsList[1])
print(getIntegerVectorAsListType)
print(arraysToString(getIntegerVectorAsList))
if(getIntegerVectorAsListType ~= "number") then
	error("getIntegerVectorAsList not Integer!")
end
callback("setIntegerVectorAsList", getIntegerVectorAsList)

print("getIntegerMatrix")
if(getIntegerMatrix ~= nil) then
	error("getIntegerMatrix already defined!")
end
getIntegerMatrix = callback("getIntegerMatrix")
getIntegerMatrixType = type(getIntegerMatrix[1][1])
print(getIntegerMatrixType)
print(arraysToString(getIntegerMatrix))
if(getIntegerMatrixType ~= "number") then
	error("getIntegerMatrix not Integer!")
end
callback("setIntegerMatrix", getIntegerMatrix)

print("getIntegerMatrixAsList")
if(getIntegerMatrixAsList ~= nil) then
	error("getIntegerMatrixAsList already defined!")
end
getIntegerMatrixAsList = callback("getIntegerMatrixAsList")
getIntegerMatrixAsListType = type(getIntegerMatrixAsList[1][1])
print(getIntegerMatrixAsListType)
print(arraysToString(getIntegerMatrixAsList))
if(getIntegerMatrixAsListType ~= "number") then
	error("getIntegerMatrixAsList not Integer!")
end
callback("setIntegerMatrixAsList", getIntegerMatrixAsList)
